reframe/core/buildsystems.py

```diff
@@ -779,6 +782,89 @@ def generated_modules(self):
         return self._eb_modules
 
 
+class Spack(BuildSystem):
+    packages = fields.TypedField(typ.List[str])
+    environment = fields.TypedField(str, type(None))
+    upstreams = fields.TypedField(typ.List[str])
+
+    def __init__(self):
+        super().__init__()
+        self.packages = []
+        self.environment = None
+        self.upstreams = []
+        self._prefix_save = None
+
+    def _prepare_scope(self):
+        os.mkdir('rfm_scope')
+        scope_config = {
+            'config': {
+                'install_tree': 'spack/opt/spack',
```

Comment:
Having only the `install_tree` and the `module_roots` is not enough to allow for a proper parallel execution of Spack. The builds can be blocked due to filesystem locks.
One still needs to add the `source_cache` and `misc_cache` entries. And they should point to unique paths for that particular Spack instance. There way we will run completely independent Spack instances. For example entries for these variables, please take a look at https://github.com/eth-cscs/production/blob/6d1dca77df8b7926a63cd5a0a39e063bb9f8b815/spack/daint/config.yaml#L19 and https://github.com/eth-cscs/production/blob/6d1dca77df8b7926a63cd5a0a39e063bb9f8b815/spack/daint/config.yaml#L22